File: backend/app/testing/books.py
from random import sample
from app.models.books import LibroCrear
from app.testing.image import image_url
def create_book(faker, cantidadUsuarios: int, seed_generos: list[str]) -> LibroCrear:
    return LibroCrear(
        nombre=faker.sentence(nb_words=3),  # Random book name (title)
        isbn=faker.unique.isbn13(),  # Random unique ISBN
        fecha_publicacion=faker.date_this_century(),  # Random publication date
        descripcion=faker.text(max_nb_chars=500),  # Random description
        portada=image_url(),  # Random image URL for cover
        id_autor=faker.random_int(min=1, max=cantidadUsuarios),  # Random author ID
        generos=sample(seed_generos, k=faker.random_int(min=1, max=3))  # Random genres (1-3)
    )

File: frontend/src/models/books.ts
import { z } from 'zod';
import { genreSchema } from './genres';

const bookBaseSchema = z.object({
	nombre: z.string().min(2),
	descripcion: z.string().min(10)
});

export const bookCreateSchema = bookBaseSchema.extend({
	isbn: z.string().min(10).max(13),
	fecha_publicacion: z.date(),
	portada: z.string().url(),
	generos: z.array(z.string()),
	id_autor: z.number()
});

export const bookSchema = bookCreateSchema.extend({
	id: z.number().positive(),
	generos: z.array(genreSchema),
	imagenes_ilustrativas: z
		.object({
			url: z.string(),
			id: z.number()
		})
		.array()
});

export const bookEditSchema = bookBaseSchema.extend({
	portada: z.string().url().nullable().optional(),
	imagenes_ilustrativas: z.array(z.string().nullable().optional()).min(1).default([''])
});

export type BookCreate = z.infer<typeof bookCreateSchema>;

export type Book = z.infer<typeof bookSchema>;

export type BookEdit = z.infer<typeof bookEditSchema>;
